src/plugins/workspace/public/workspace_client.ts

```diff
@@ -0,0 +1,275 @@
+    const currentWorkspaceId = this.workspaces.currentWorkspaceId$.getValue();
+    if (!currentWorkspaceId) {
+      return {
+        success: false,
+        error: 'You are not in any workspace yet.',
+      };
+    }
+
+    return {
+      success: true,
+      result: currentWorkspaceId,
+    };
+  }
+
+  /**
+   * Do a find in the latest workspace list with current workspace id
+   */
+  public async getCurrentWorkspace(): Promise<IResponse<WorkspaceAttribute>> {
+    const currentWorkspaceIdResp = await this.getCurrentWorkspaceId();
+    if (currentWorkspaceIdResp.success) {
+      const currentWorkspaceResp = await this.get(currentWorkspaceIdResp.result);
+      return currentWorkspaceResp;
+    } else {
+      return currentWorkspaceIdResp;
+    }
+  }
+
+  /**
+   * Create an workspace
+   *
+   * @param attributes
+   * @returns
+   */
+  public async create(
+    attributes: Omit<WorkspaceAttribute, 'id'>
+  ): Promise<IResponse<WorkspaceAttribute>> {
+    const path = this.getPath();
+
+    const result = await this.safeFetch<WorkspaceAttribute>(path, {
+      method: 'POST',
+      body: JSON.stringify({
+        attributes,
+      }),
+    });
+
+    if (result.success) {
+      await this.updateWorkspaceList();
+    }
+
+    return result;
+  }
+
+  /**
+   * Deletes a workspace by workspace id
+   *
+   * @param id
+   * @returns
+   */
+  public async delete(id: string): Promise<IResponse<null>> {
+    const result = await this.safeFetch<null>(this.getPath(id), { method: 'DELETE' });
+
+    if (result.success) {
+      await this.updateWorkspaceList();
+    }
+
+    return result;
+  }
+
+  /**
+   * Search for workspaces
+   *
+   * @param {object} [options={}]
+   * @property {string} options.search
+   * @property {string} options.searchFields - see OpenSearch Simple Query String
+   *                                        Query field argument for more information
+   * @property {integer} [options.page=1]
+   * @property {integer} [options.perPage=20]
+   * @property {array} options.fields
+   * @returns A find result with workspaces matching the specified search.
+   */
+  public list(
+    options?: WorkspaceFindOptions
+  ): Promise<
+    IResponse<{
+      workspaces: WorkspaceAttribute[];
+      total: number;
+      per_page: number;
+      page: number;
+    }>
+  > {
+    const path = this.getPath('_list');
+    return this.safeFetch(path, {
+      method: 'POST',
+      body: JSON.stringify(options || {}),
+    });
+  }
+
+  /**
+   * Fetches a single workspace by a workspace id
+   *
+   * @param {string} id
+   * @returns The workspace for the given id.
+   */
+  public get(id: string): Promise<IResponse<WorkspaceAttribute>> {
+    const path = this.getPath(id);
+    return this.safeFetch(path, {
+      method: 'GET',
+    });
+  }
+
+  /**
+   * Updates a workspace
+   *
+   * @param {string} id
+   * @param {object} attributes
+   * @returns
+   */
+  public async update(
```

Comment:
can we finish documenting this function?

Reply:
Sure, done for that.